extensions/iam/distributed-identity/identity-did-web/src/test/java/org/eclipse/dataspaceconnector/iam/did/web/resolution/WebDidResolverTest.java

```diff
@@ -0,0 +1,98 @@
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+package org.eclipse.dataspaceconnector.iam.did.web.resolution;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import okhttp3.Interceptor;
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import okhttp3.Response;
+import okhttp3.ResponseBody;
+import org.easymock.EasyMock;
+import org.eclipse.dataspaceconnector.spi.monitor.Monitor;
+import org.jetbrains.annotations.NotNull;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+
+import static okhttp3.Protocol.HTTP_1_1;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Fail.failBecauseExceptionWasNotThrown;
+
+/**
+ * Verifies Web DID resolution.
+ */
+class WebDidResolverTest {
+
+    @Test
+    void verifyResolveDocumentIsSuccessful() {
+        var interceptor = new Interceptor() {
+            @NotNull
+            @Override
+            public Response intercept(@NotNull Interceptor.Chain chain) throws IOException {
+                var didStream = Thread.currentThread().getContextClassLoader().getResourceAsStream("did.json");
+                assert didStream != null;
+                var didDocument = new String(didStream.readAllBytes(), StandardCharsets.UTF_8);
+                var body = ResponseBody.create(didDocument, MediaType.get("application/json"));
+                return new Response.Builder().body(body).protocol(HTTP_1_1).request(chain.request()).code(200).message("ok").build();
+            }
+        };
+        var resolver = createResolver(interceptor);
+        var document = resolver.resolve("did:web:foo.com:edc:EiDfkaPHt8Yojnh15O7egrj5pA9tTefh_SYtbhF1-XyAeA");
+        assertThat(document).isNotNull();
+    }
+
+    @Test
+    void verifyResolveDocumentNotFound() {
+        var interceptor = new Interceptor() {
+            @NotNull
+            @Override
+            public Response intercept(@NotNull Interceptor.Chain chain) {
+                var body = ResponseBody.create("", MediaType.get("application/json"));
+                return new Response.Builder().body(body).protocol(HTTP_1_1).request(chain.request()).code(404).message("notfound").build();
+            }
+        };
+        var resolver = createResolver(interceptor);
+        var document = resolver.resolve("did:web:foo.com:edc:EiDfkaPHt8Yojnh15O7egrj5pA9tTefh_SYtbhF1-XyAeA");
+        assertThat(document).isNull();
+    }
+
+    @Test
+    void verifyConvertToUrl() throws Exception {
+        var resolver = createResolver();
+
+        assertThat(resolver.keyToUrl("did:web:w3c-ccg.github.io")).isEqualTo("https://w3c-ccg.github.io/.well-known/did.json");
```

Comment:
I think testing private methods (actually this is a default one, but I think it has this visibility just for testing purposes) is not a good practice, in this case we should use the interceptor way to test that or extract the transformation logic into another component. I think the latter is better

Reply:
I'm not too hung up on that, I think it is OK to test non-public methods but we can discuss that in another thread.

I agree it's much better, in this case, to separate it out, so I went ahead and did that.